File: src/main/java/dondeInvierto/dominio/repo/RepositorioMetodologias.java
package dondeInvierto.dominio.repo;

import dondeInvierto.dominio.metodologias.Metodologia;
import dondeInvierto.dominio.repo.AbstractLocalRepository;

import javax.persistence.Entity;
import java.util.List;

@Entity
public class RepositorioMetodologias extends AbstractLocalRepository<Metodologia> {

	public void agregarMetodologias(List<Metodologia> metodologias) {
		metodologias.forEach(metodologia -> agregar(metodologia));
	}
	
	@Override
	protected String mensajeEntidadExistenteError(Metodologia elemento) {
		return "Ya existe una metodología con el nombre " + elemento.getNombre();
	}

	public Metodologia obtenerPorId(Long id) {
		return obtenerTodos().stream().filter(metodologia -> metodologia.getId().equals(id)).findFirst()
				.orElseThrow(null);//Estoy seguro que la metodologia va a existir
	}

}

File: src/main/java/dondeInvierto/dominio/repo/RepositorioMetodologias.kt
package dondeInvierto.dominio.repo

import dondeInvierto.dominio.metodologias.Metodologia
import javax.persistence.Entity

@Entity
class RepositorioMetodologias : AbstractLocalRepository<Metodologia>() {

    override fun mensajeEntidadExistenteError(elemento: Metodologia) = "Ya existe una metodología con el nombre " + elemento.nombre

    fun obtenerPorId(id: Long?): Metodologia = obtenerTodos().first { it.id == id }

}